### DISCUSSION TITLE
Wavelet Coherence application

### DISCUSSION THREAD

**Discussion by thelightonmyway** [LinkedEarth/Pyleoclim_util#627] [Q&A]
Hi,

I could use `wavelet_coherence` to do some wavelet analysis thanks for `Pyleoclim`. And I get some trouble when I'm reproducing the results of the paper from [https://doi.org/10.1029/2023GL103323](https://github.com/LinkedEarth/Pyleoclim_util/issues/url). I'm grateful to @CommonClimate for help and I could set the parameters same as the paper now.

However, there still seems some difference between my reproduction and the paper. It could be seen as:
<img width="869" alt="cd692839ef3a499a3cb37c3ed005097" src="https://github.com/user-attachments/assets/6de0607f-7ba9-4253-a5eb-21a5d84b3773">
the left is from the paper and the right is my result.

The main character is similar in periods 8-16 months compared with the origin, but it appears difference in other periods, such as:
1. the energy of mine is lower;
2.  the arrow of mine is inconsistent.
This is my result without significance, in order to see the arrow more intuitively.
<img width="465" alt="5762430eb77aa481e7826edca67c39e" src="https://github.com/user-attachments/assets/8caa0485-df00-4090-a285-3be3d6344cb8">

I'm sure that the method and data I use is correct, since it's not complex with the result of the coherence wavelet does not change with the data being anomaly or original.

For reference, the data used could be get from:
[MonthlyAnomSWS_IP.txt](https://github.com/user-attachments/files/17731098/MonthlyAnomSWS_IP.txt)
[pv.txt](https://github.com/user-attachments/files/17731125/pv.txt)

What troubles me is that:
1. How could I draw the COI completely? since my COI seems not complete now.
2. What causes the difference between my result and the paper?

Any reply will be appreciated sincerely!

**Comment by CommonClimate** [comment]
Hi @thelightonmyway 
1. you can tweak the frequency vector so it shows very long scales. Naturally, that is a bit of a waste since any wavelet coefficient under the cone of influence is, by definition, suspect. 
2. That is for you to find out! Different packages implement the same method(s) in subtly different ways, and much depends on the parameters being chosen. Are the results qualitatively inconsistent? 
The default frequency method used in wavelet analysis in Pyleoclim is `log`, so that is one first source of potential difference to explore.

**Reply by thelightonmyway** [reply]
I see...thank you very much! And how could I change the frequency method of `log`? I means, how to alter it when I have set the `frequency vector` as `freq_vector_scale(time,dj=1/12,s0=2,j1=83,mother='MORLET',param=6)`?

**Comment by CommonClimate** [comment]
Sorry, I meant in general that the defaults in Pyleoclim were different from Aslak Grinsted's code. In this case, your frequency vector should be getting somewhat closer, but again there is no way to know unless you do a side-by-side comparison, and that's not our job! Good luck and please let us know if you have other questions.

**Reply by thelightonmyway** [reply]
Well, I got it, thank you for your help!
